### ISSUE TITLE
Postgres API: Enable shortcut for selecting all credential types

### ISSUE THREAD

**Issue by siuc-nate** [CredentialEngine/CredentialRegistry#493]
In order to avoid needing to list all 20+ credential types in a query, and possibly for other purposes, we need a way to use a "shortcut" or stand-in token that gets translated to the full list of types, and ensure that that list stays synchronized with the schema as it changes.

In the SPARQL implementation, such a shortcut currently exists for credential types, but I tried to come up with a naming scheme that could apply to other sets of classes as well:

`search:ClassGroup_Credential`

I'm open to changing that token to something else if something else makes more sense, but the functionality involved is that it starts with `ceterms:Credential` and crawls the schema to look for all subclasses (and subclasses of those subclasses, and so on) of that class, until it arrives at the full list of all credential types. That full list then gets substituted in place of that token before the rest of the query is processed.

That basically turns this query:
```
{
	"@type": [
		"search:ClassGroup_Credential"
	],
	"ceterms:name": "some credential name",
	"etc": "etc"
}
```
into this query:
```
{
	"@type": [
		"ceterms:ApprenticeshipCertificate",
		"ceterms:AssociateDegree",
		"ceterms:BachelorDegree",
		"ceterms:Badge",
		"ceterms:Certificate",
		"ceterms:CertificateOfCompletion",
		"ceterms:Certification",
		"ceterms:Degree",
		"ceterms:DigitalBadge",
		"ceterms:Diploma",
		"ceterms:DoctoralDegree",
		"ceterms:GeneralEducationDevelopment",
		"ceterms:JourneymanCertificate",
		"ceterms:License",
		"ceterms:MasterCertificate",
		"ceterms:MasterDegree",
		"ceterms:MicroCredential",
		"ceterms:OpenBadge",
		"ceterms:ProfessionalDoctorate",
		"ceterms:QualityAssuranceCredential",
		"ceterms:ResearchDoctorate",
		"ceterms:SecondarySchoolDiploma"
	],
	"ceterms:name": "some credential name",
	"etc": "etc"
}
```

There are other places in the schema where this might come in handy, such ceterms:Organization and its subclasses, ceterms:PathwayComponent subclasses, and probably others - so the approach should be flexible enough to be extended in the future, but for now, we just need it for credential types.

As I think about it more, it would probably be wise to namespace the class, so perhaps something like `search:ClassGroup_ceterms:Credential` instead would be better. I'm open to suggestions.

**Comment by jeannekitchens** [comment]
@siuc-nate this needs to be implemented. It already is in place in the existing search so what needs to be done to implement?  If you're not handling this, I need confirmation that Tapestry has what they need from the current search to do so.

**Comment by siuc-nate** [comment]
We just need to make sure we're on the same page as far as the syntax of it (to future-proof it for other shortcuts like it) and then the registry team will implement the transformation.

**Comment by siuc-nate** [comment]
We will come up with a list of additional groups of classes that make sense for this kind of shortcut process.

**Comment by neaf** [comment]
@siuc-nate The changes are deployed to sandbox. Here's what's happening...

I'm fetching following 3 schemas:
https://credreg.net/ctdl/schema
https://credreg.net/ctdlasn/schema
https://credreg.net/qdata/schema

I'm looking at all `rdfs:Class` elements and generate a tree based on their `rdfs:subClassOf` propery.

Current tree can be found here: https://github.com/CredentialEngine/CredentialRegistry/blob/825666a828d9830306aece3fccec5a6cf47b887a/fixtures/subclasses_map.json

There's also a script that updates the tree but at this point it requires running from dev machine and committing changes into the repo.

We already had a query value extension that we used for text matching types. I figured it fit greatly to what we're doing here so I used it:

```
{
  "@type": {
		"search:value": "ceterms:Badge",
		"search:matchType": "search:subClassOf"
	}
}

{
  "@type": {
		"search:value": ["ceterms:Credential", "ceterms:Collection"],
		"search:matchType": "search:subClassOf"
	}
}

Please let me know if it satisfies the requirements and works as you expect so we can deploy to production.

Cheers 🙇‍♂️
